### PR TITLE
#109 Adding Preview support for Master Slide backgrounds

### PR DESCRIPTION
**Pull request by ksk385**

### CODE REVIEW

**Comment by ksk385** [comment]
Lots of the changes are formatting because I setup my prettier on vscode.

The main changes I have made are in `SlidePreview` component:

```
+
+  const backgroundColor = slide.backgroundColor ?? masterSlide?.backgroundColor;
+  const backgroundImage = slide.backgroundImage ?? masterSlide?.backgroundImage;
+
   return (
     <div
       ref={ref}
       style={{
         width: "100%",
         height: width / (dimensions[0] / dimensions[1]),
-        backgroundColor: slide.backgroundColor
-          ? normalizedColorToCSS(slide.backgroundColor)
+        backgroundColor: backgroundColor
+          ? normalizedColorToCSS(backgroundColor)
           : "white",
         backgroundImage:
-          slide.backgroundImage && slide.backgroundImage?.kind === "path"
-            ? `url("${slide.backgroundImage.path}")`
-            : `url("data:${slide.backgroundImage?.data}")`,
+          backgroundImage && backgroundImage?.kind === "path"
+            ? `url("${backgroundImage.path}")`
+            : `url("data:${backgroundImage?.data}")`,
+        backgroundSize: "contain",
```

**Comment by ksk385** [comment]
Updated the PR to include image support.

File: src/renderer.ts [hunk 1]
```diff
@@ -219,6 +219,76 @@ const renderSlide = async (
...
+        w: object.style.w,
+        h: object.style.h,
+        ...(object.style.backgroundColor && {
+          fill: { color: object.style.backgroundColor },
+        }),
+      },
+    };
+  } else if (object.kind === "image") {
+    let data = "";
+    if (object.src.kind === "data") {
```
[the remark this answers is not in the document]
Comment (ksk385) [reply src/renderer.ts:244]: Makes sense. Will update that.

**Comment by wyozi** [comment]
Thanks 👍

**Pull request merged**